src/binary/ibc-setup/commands/balances.ts

```diff
@@ -0,0 +1,74 @@
+import path from 'path';
+
+import { GasPrice } from '@cosmjs/launchpad';
+import { Decimal } from '@cosmjs/math';
+import { DirectSecp256k1HdWallet } from '@cosmjs/proto-signing';
+
+import { IbcClient } from '../../../lib/ibcclient';
+import { registryFile } from '../../constants';
+import { getDefaultHomePath } from '../utils/get-default-home-path';
+import { loadAndValidateApp } from '../utils/load-and-validate-app';
+import { loadAndValidateRegistry } from '../utils/load-and-validate-registry';
+import { resolveMnemonicOption } from '../utils/resolve-mnemonic-option';
+import { resolveOption } from '../utils/resolve-option';
+import { resolveRequiredOption } from '../utils/resolve-required-option';
+
+import { Flags, getAddresses, Options } from './keys-list';
+
+export async function balances(flags: Flags) {
+  const home = resolveRequiredOption('home')(
+    flags.home,
+    process.env.RELAYER_HOME,
+    getDefaultHomePath
+  );
+  const app = loadAndValidateApp(home);
+
+  const keyFile = resolveOption(
+    flags.keyFile,
+    app?.keyFile,
+    process.env.KEY_FILE
+  );
+
+  const options: Options = {
+    home,
+    mnemonic: await resolveMnemonicOption({
+      interactive: flags.interactive,
+      mnemonic: flags.mnemonic,
+      keyFile: keyFile,
+      app,
+    }),
+  };
+
+  await run(options);
+}
+
+export async function run(options: Options) {
+  const registryFilePath = path.join(options.home, registryFile);
+  const registry = loadAndValidateRegistry(registryFilePath);
+
+  const addresses = await getAddresses(registry.chains, options.mnemonic);
+
+  for (const [chain, data, address] of addresses) {
+    const signer = await DirectSecp256k1HdWallet.fromMnemonic(
+      options.mnemonic,
+      undefined,
```

Comment:
Should we provide `hd_path` here?

Reply:
yes

Reply:
https://github.com/confio/ts-relayer/pull/91/files#diff-58673c679ff81312a0f7389a4cf697deac398637a352094bd9d4832a5cf28ee4R49